test/net/sourceforge/kolmafia/textui/RuntimeLibraryTest.java

```diff
@@ -1384,12 +1397,57 @@ void trackedByIncludesAllTracks() {
             actual,
             equalTo(
                 """
-            Returned: aggregate string [3]
-            0 => Transcendent Olfaction
-            1 => Gallapagosian Mating Call
-            2 => Red-Nosed Snapper
-            """));
+                                Returned: aggregate string [3]
+                                0 => Transcendent Olfaction
+                                1 => Gallapagosian Mating Call
+                                2 => Red-Nosed Snapper
+                                """));
       }
     }
   }
+
+  /**
+   * This test is intended to test running the maximizer from a command or script (not the GUI),
+   * show that the maximizer chooses a weapon and then emits the commands to equip that weapon.
+   */
+  @Test
+  public void itShouldMaximizeAndEquipSelectedWeapon() {
+    String maxStr = "effective";
+    HttpClientWrapper.setupFakeClient();
+    var cleanups =
+        new Cleanups(
+            withStats(10, 5, 5),
+            withEquippableItem("seal-skull helmet"),
+            withEquippableItem("astral shirt"),
+            withEquippableItem("old sweatpants"),
+            withEquippableItem("sewer snake"),
+            withEquippableItem("seal-clubbing club"));
+    String out;
+    String cmd;
+    cmd = "maximize(\"" + maxStr + "\", false)";
+    try (cleanups) {
+      out = execute(cmd);
+    }
+    assertFalse(out.isEmpty());
+    assertTrue(out.contains("Putting on seal-skull helmet..."));
+    assertTrue(out.contains("Wielding seal-clubbing club..."));
+    assertTrue(out.contains("Putting on old sweatpants..."));
+    assertContinueState();
+    var requests = getRequests();
+    assertTrue(requests.size() > 1);
```

Comment:
```suggestion
    assertTrue(requests.size() >= 1);
```

Reply:
That's a hangover from a case when there were several requests and the first one had nothing to do with the weapon.  In the new context given the following iteration the check should be for any number of requests.  Stand by.